lib/deploy/functions/release.js

```diff
@@ -110,123 +117,53 @@ module.exports = function(context, options, payload) {
+    } else {
+      interval = 5 * 1000;
     }
-
-    function logSuccessfulOps(ops) {
-      var functionUrls = [];
-      _.forEach(ops, function(operation) {
-        var parts = operation.func.split('/');
-        var functionName = parts[5];
-        if (operation.triggerUrl && operation.type !== 'delete') {
-          var entry = { 'funcName': functionName, 'url': operation.triggerUrl};
-          functionUrls = _.concat(functionUrls, entry);
-        }
-        utils.logSuccess(chalk.bold.green('functions[' + functionName + ']: ') + 'Successful ' + operation.type + ' operation. ');
-      });
-
-      if (functionUrls.length > 0) {
-        _.set(context, 'functionUrls', functionUrls);
+    var pollFunction = gcp.cloudfunctions.check;
+    var printSuccess = function(op) {
+      _endTimer(op.functionName);
+      utils.logSuccess(chalk.bold.green('functions[' + op.functionName + ']: ') + 'Successful ' + op.type + ' operation. ');
+      if (op.triggerUrl && op.type !== 'delete') {
+        logger.info(chalk.bold('Function URL'), '(' + op.funcName + '):', op.triggerUrl);
       }
-    }
-
-    var failedChecks = _.filter(successfulCalls, function(op) {
-      return !op.done && op.error;
-    });
-    var failedOps = _.filter(successfulCalls, function(op) {
-      return op.done && op.error;
-    });
-    var successfulOps = _.filter(successfulCalls, function(op) {
-      return op.done && !op.error;
-    });
-
-    var failed = false;
-
-    var failCount = failedOps.length + failedCalls.length;
-    track('Functions Deploy (Result)', 'failure', failCount.length || 0);
-    track('Functions Deploy (Result)', 'success', successfulOps.length || 0);
-
-    if (failedCalls.length > 0) {
-      failed = true;
-      utils.logWarning(chalk.bold.yellow('functions: ') + failedCalls.length + ' function(s) failed to be deployed.');
-      logger.debug(failedCalls);
-    }
-    if (failedOps.length > 0) {
-      failed = true;
-      logFailedOps(failedOps);
-    }
-    if (failedChecks.length > 0) {
-      utils.logWarning(chalk.bold.yellow('functions: ') + 'failed to get status of ' + failedChecks.length + ' operation(s).');
-      utils.logWarning(chalk.bold.yellow('functions: ') + 'run ' + chalk.bold('firebase functions:log') + ' in a few minutes to ensure that your functions deployed properly.');
-    }
-    if (successfulOps.length > 0) {
-      logSuccessfulOps(successfulOps);
-      if (!failed) {
-        utils.logSuccess(chalk.bold.green('functions: ') + 'all functions deployed successfully!');
+    };
+    var printFail = function(op) {
+      _endTimer(op.functionName);
+      failedDeployments += 1;
+      utils.logWarning(chalk.bold.yellow('functions[' + op.functionName + ']: ') + 'Deployment error.');
+      if (op.error.code === 8) {
+        logger.debug(op.error.message);
+        logger.info('You have exceeded your deployment quota, please deploy your functions in batches by using the --only flag, ' +
```

Comment:
I wish the CLI could just emit a warning that the quota has been hit, pause 100s, and try the next batch.